src/util/uri.cpp

```diff
@@ -79,7 +80,9 @@ static inline util::sview updated_copy(const std::vector<char>& to_copy,
                                        util::csview view,
                                        const std::vector<char>& from_copy)
 {
-  return {to_copy.data() + (view.data() - from_copy.data()), view.size()};
+  // sometimes the source is empty, but we need a valid empty string
+  if (view.data() == nullptr) return {&to_copy.back(), 0};
```

Comment:
next line with indent ?
avoids accidentally reading the next line as how the case is handled

Reply:
This bugfix is already on master, but we can change at least change the future. PRs welcome!